### PR TITLE
Graticule integration

### PR DESCRIPTION
**Pull request by mridulnagpal**

### CODE REVIEW

**Comment by jywarren** [comment]
Hi, Mridul - when it's running, can you push a copy to your gh-pages branch too so we can try it out? Thanks!

**Comment by mridulnagpal** [comment]
Here you go https://mridulnagpal.github.io/leaflet-blurred-location/examples/

**Comment by jywarren** [comment]
This is looking really good. We need to make a couple follow up issues:

1. let's tweak the grid resolution per zoom level so that multiple grid squares are shown for each zoom level (this may take some manual tweaking) -- and this will be what sets #44's `setZoomByPrecision()` - even if we don't do an algorithmic approach, we could actually literally do a look up table... worth considering. This can be a separate issue and separate PR.
2. we need to highlight the center grid square now (I think this'll be the hard part, but if you look in how Graticule is implemented, it shouldn't be too hard). This can be in a separate issue.
3. Can you require the Graticule plugin via npm using your `package.json` ?, instead of directly including the source? check if it's available on npmjs.org. This should be done in this PR. Thank you!

Thank you, Mridul!!! This is great progress.

**Comment by mridulnagpal** [comment]
I don't find it on npm I tried earlier as well that's why we need it in source, also we will be altering the code for graticulate, thus we need it anyways.

**Comment by mridulnagpal** [comment]
I think this is only appllicable to npm packages that are registered and this one is not. I am getting several errors. I tried all of the solutions on that page

**Comment by jywarren** [comment]
Can you post what you tried? I'm sure this works without npm registration; see how the PublicLab.Editor is doing this: https://github.com/publiclab/PublicLab.Editor/blob/master/package.json#L28

I had to fork and adjust that library, so i linked directly to the URL of my modded fork, specifying a branch with `#branchname`.

**Comment by mridulnagpal** [comment]
I did `npm install --save-dev git://github.com/Leaflet/Leaflet.Graticule.git#master` and got error 
image

**Comment by jywarren** [comment]
Try adding this to `package.json` --

```js
"leaflet-graticule": "git://github.com/Leaflet/Leaflet.Graticule.git#master"
```
then run `npm update`

**Comment by mridulnagpal** [comment]
I actually tried that earlier, same issue appears to occur again 
image

**Comment by jywarren** [comment]
Ack! it's because it's `git://github.com/Leaflet/Leaflet.Graticule.git`, not `git://github.com/Leaflet/Graticule.git`! That really got me. 

But also it needs a package.json -- so i forked, opened a new branch & PR, and you can now:

```
"leaflet-graticule": "git://github.com/jywarren/Leaflet.Graticule.git#patch-1"
```

until they accept my PR: https://github.com/Leaflet/Leaflet.Graticule/pull/1

**Comment by mridulnagpal** [comment]
My bad, corrected this

File: spec/javascripts/test_spec.js [hunk 1]
```diff
@@ -29,19 +29,19 @@ describe("Basic testing", function() {
...
+  //   blurredLocation.setZoom(13);
+  //
+  //   expect(blurredLocation.gridSystem.getCellSize().rows).toBe(58.25);
+  //   expect(blurredLocation.gridSystem.getCellSize().cols).toBe(94.63);
+  //
+  //   blurredLocation.setZoom(10);
+  //
+  //   expect(blurredLocation.gridSystem.getCellSize().rows).toBe(72.8);
+  //   expect(blurredLocation.gridSystem.getCellSize().cols).toBe(118.3);
+  //
```
Comment (jywarren) [inline spec/javascripts/test_spec.js]: Is there a way to write a test for this? Or shall we just depend on Graticule? I'm not sure, does Graticule have tests?

File: src/blurredLocation.js [hunk 2]
```diff
@@ -3,9 +3,16 @@ BlurredLocation = function BlurredLocation(options) {
...
 
   options = options || {};
-  options.map = options.map || L.map('map');
+  options.location = options.location || {
+    lat: 41.011234567,
+    lon: -85.66123456789
+  };
+
+  options.zoom = options.zoom || 6;
+  options.map = options.map || new L.Map('map',{zoomControl:false}).setView([options.location.lat, options.location.lon], options.zoom);
```
Comment (jywarren) [inline src/blurredLocation.js]: Can you break this up onto multiple lines? For readability.

File: src/blurredLocation.js [hunk 3]
```diff
@@ -26,15 +33,9 @@ BlurredLocation = function BlurredLocation(options) {
 
   Interface = options.Interface(InterfaceOptions);
 
-  L.tileLayer("https://a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png").addTo(options.map);
-
-  options.location = options.location || {
-    lat: 41.011234567,
-    lon: -85.66123456789
-  };
+  var stamenTerrain = L.tileLayer("https://a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png").addTo(options.map);
```
Comment (jywarren) [inline src/blurredLocation.js]: Hmm, why this name? Maybe something more generic, like tileLayer?

File: src/core/gridSystem.js [hunk 4]
```diff
@@ -2,167 +2,33 @@ module.exports = function gridSystem(options) {
...
-    return new L.VirtualGrid(options);
-  };
-
-  var layer = L.virtualGrid({
-                cellSize: { rows:100, cols:100 }
-              }).addTo(map);
+  function addGrid() {
+     layer = L.latlngGraticule({
+                    showLabel: true,
+                    zoomInterval: [
```
Comment (jywarren) [inline src/core/gridSystem.js]: Is this repeated? Maybe use an options parameter?

File: spec/javascripts/test_spec.js (hunk 1 above)
Comment (mridulnagpal) [inline spec/javascripts/test_spec.js]: I think we will have to depend on graticule for that

File: src/blurredLocation.js (hunk 2 above)
Comment (mridulnagpal) [inline src/blurredLocation.js]: Sure thing

File: src/blurredLocation.js (hunk 3 above)
Comment (mridulnagpal) [inline src/blurredLocation.js]: It was just a default, will change to tileLayer

File: src/core/gridSystem.js (hunk 4 above)
Comment (mridulnagpal) [inline src/core/gridSystem.js]: Done

**Comment by mridulnagpal** [comment]
Done Please have a look.

**Comment by jywarren** [comment]
Great!!!

**Pull request merged**